File: src/main/java/entities/gestionSuivi/Planning.java
package entities.gestionSuivi;

public class Planning {
    private int id_Planning;
    private int idObjectif;
    private  int id;
    private int idCoach;
    private String trainingProg ;

    private String foodProg ;

    public Planning(){

    }

    public Planning(int id_Objectif,int id_Coach,String trainningProg,String foodProg){
        this.setIdObjectif(id_Objectif);
        this.setIdCoach(id_Coach);
        this.setTrainingProg(trainningProg);
        this.setFoodProg(foodProg);

    }

    public Planning(int id_Planning,int id_Objectif,int id_Coach,String trainningProg,String foodProg){
        this.setId_Planning(id_Planning);
        this.setIdObjectif(id_Objectif);
        this.setIdCoach(id_Coach);
        this.setTrainingProg(trainningProg);
        this.setFoodProg(foodProg);

    }



    public int getId_Planning() {
        return id_Planning;
    }

    public void setId_Planning(int id_Planning) {
        this.id_Planning = id_Planning;
    }

    public int getIdObjectif() {
        return idObjectif;
    }

    public void setIdObjectif(int idObjectif) {
        this.idObjectif = idObjectif;
    }

    public int getIdCoach() {
        return idCoach;
    }

    public void setIdCoach(int idCoach) {
        this.idCoach = idCoach;
    }

    public String getTrainingProg() {
        return trainingProg;
    }

    public void setTrainingProg(String trainingProg) {
        this.trainingProg = trainingProg;
    }

    public String getFoodProg() {
        return foodProg;
    }

    public void setFoodProg(String foodProg) {
        this.foodProg = foodProg;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: webapp/src/GymPlus/src/Entity/Planning.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\PlanningRepository;

use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: PlanningRepository::class)]
class Planning
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idPlanning;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Field is required")]

    private ?string $trainingprog;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Field is required")]
    private ?string $foodprog;


    #[ORM\ManyToOne(targetEntity: Objectif::class)]
    #[ORM\JoinColumn(name: "idobjectif", referencedColumnName: "idobjectif")]
    private ?Objectif $idobjectif;

    public function getIdPlanning(): ?int
    {
        return $this->idPlanning;
    }

    public function getTrainingprog(): ?string
    {
        return $this->trainingprog;
    }

    public function setTrainingprog(?string $trainingprog): static
    {
        $this->trainingprog = $trainingprog;

        return $this;
    }

    public function getFoodprog(): ?string
    {
        return $this->foodprog;
    }

    public function setFoodprog(?string $foodprog): static
    {
        $this->foodprog = $foodprog;

        return $this;
    }

    public function getIdobjectif(): ?Objectif
    {
        return $this->idobjectif;
    }

    public function setIdobjectif(?Objectif $idobjectif): static
    {
        $this->idobjectif = $idobjectif;

        return $this;
    }


}
